zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -239,16 +250,8 @@ protected ZkClient(IZkConnection zkConnection, int connectionTimeout, long opera
     } else {
       LOG.info("ZkClient monitor key or type is not provided. Skip monitoring.");
     }
-
-    connect(connectionTimeout, this);
-
-    try {
-      if (_monitor != null) {
-        _monitor.register();
-      }
-    } catch (JMException e){
-      LOG.error("Error in creating ZkClientMonitor", e);
-    }
+    // set the zookeeper watcher for notification, use "this" native zkclient instance by default if not explicitly set
+    _watcher = watcher == null ? this : watcher;
```

Comment:
This changes the behavior, right?

Reply:
No, the current behavior remains the same